messaging/src/main/java/org/axonframework/common/infra/FilesystemStyleComponentDescriptor.java

```diff
@@ -0,0 +1,377 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.common.infra;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.configuration.Component;
+
+import java.util.*;
+
+/**
+ * A {@link ComponentDescriptor} implementation inspired by filesystem structures. Components are represented as
+ * "directories" with properties as "files" and references as "symbolic links". This creates a clear hierarchical
+ * representation with explicit paths to show component relationships.
+ * <p>
+ * Each component is assigned a path in the virtual filesystem. References to components that have already been
+ * described are represented as symbolic links, indicated with the {@code -> /path/to/component} syntax, preventing
+ * infinite recursion in circular references.
+ * <p>
+ * Example output:
+ * <pre>
+ * /
+ * ├── systemComponent/
+ * │   ├── _ref: 123512
+ * │   ├── _type: EventProcessor
+ * │   ├── name: my-processor
+ * │   └── configuration/
+ * │       ├── _ref: 123513
+ * │       ├── _type: ProcessorConfiguration
+ * │       ├── batchSize: 100
+ * │       └── processor -> /systemComponent
+ * </pre>
+ *
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ */
+public class FilesystemStyleComponentDescriptor implements ComponentDescriptor {
+
+    private static final String ROOT_PATH = "/";
+    private static final String PATH_SEPARATOR = "/";
+
+    private final Map<Object, String> componentPaths;
+    private final Map<String, Object> properties;
+    private final String currentPath;
+
+    /**
+     * Constructs a new {@code FilesystemComponentDescriptor} as the root of the hierarchy.
+     */
+    public FilesystemStyleComponentDescriptor() {
+        this(new IdentityHashMap<>(), ROOT_PATH);
+    }
+
+    /**
+     * Private constructor used for creating nested descriptors that share the component paths map.
+     *
+     * @param componentPaths Map containing paths to already processed components.
+     * @param currentPath    The current path in the virtual filesystem.
+     */
+    private FilesystemStyleComponentDescriptor(Map<Object, String> componentPaths, String currentPath) {
+        this.componentPaths = componentPaths;
+        this.currentPath = currentPath;
+        this.properties = new LinkedHashMap<>();
+    }
+
+    @Override
+    public void describeProperty(@Nonnull String name, Object object) {
+        if (object instanceof DescribableComponent component) {
+            describeComponent(name, component);
+        } else {
+            properties.put(name, object);
+        }
+    }
+
+    private void describeComponent(String name, DescribableComponent component) {
+        var componentPath = currentPathChild(name);
```

Comment:
Super duper nit: why make the `componentPath` at the beginning of this method, while it might not be used if the given `component` is already contained in the `componentPaths` collection? Thus, you could also construct the `componentPath` after the if-block, for example.